examples-gallery/intermediate/plot_countersteer.py

```diff
@@ -0,0 +1,290 @@
+formulation from [Karnopp2004]_ here.
+
+The goal of this optimal control problem is to find the steering control input
+to make a 90 degree right-hand turn in minimal time with limits on steer and
+roll angular rate while traveling at 18 km/h.
+
+.. _countersteering: https://en.wikipedia.org/wiki/Countersteering
+
+"""
+from opty import Problem
+from opty.utils import MathJaxRepr
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+import numpy as np
+import sympy as sm
+import sympy.physics.mechanics as me
+
+# %%
+# Specify the equations of motion
+# -------------------------------
+#
+# The model is constructed using several constant parameters:
+#
+# - :math:`h`: distance mass center is from the ground contact line
+# - :math:`a`: longitudinal distance of the mass center from the rear contact
+# - :math:`b`: wheelbase length
+# - :math:`v`: constant longitudinal speed at the rear contact
+# - :math:`g`: acceleration due to gravity
+# - :math:`m`: mass of bicycle and rider
+# - :math:`I_1`: roll principle moment of inertia
+# - :math:`I_2`: pitch principle moment of inertia
+# - :math:`I_3`: yaw principle moment of inertia
+h, a, b, v, g = sm.symbols('h a, b, v, g', real=True)
+m, I1, I2, I3 = sm.symbols('m, I1, I2, I3', real=True)
+
+# %%
+# The essential dynamics are described by a single degree of freedom model with
+# roll angle :math:`\theta` and its angular rate :math:`\dot{\theta}` being the
+# essential state variables. The location of the rear contact in the ground
+# plane :math:`x,y` and heading :math:`\psi` will also be tracked. The input is
+# the steering angle :math:`\delta`.
+theta, thetadot = me.dynamicsymbols('theta, thetadot', real=True)
+x, y, psi = me.dynamicsymbols('x, y, psi', real=True)
+delta, deltadot = me.dynamicsymbols('delta, deltadot', real=True)
+t = me.dynamicsymbols._t
+
+# %%
+# The first two differential equations are the essential single degree of
+# freedom dynamics followed by the differential equations to track :math:`x,y`
+# and :math:`\psi`. Both :math:`\delta` and :math:`\dot{\delta}` are present as
+# inputs to the dynamics. In the optimal control problem, both of these input
+# trajectories are sought with the constraint that :math:`d\delta/dt` holds. To
+# manage this with opty, add a differential equation that ensures steering
+# angle and steering rate variables are related by time differentiation and
+# make :math:`\delta` a pseudo state variable with the highest derivative
+# :math:`\dot{\delta}` being the single unknown input trajectory.
+eom = sm.Matrix([
+    theta.diff(t) - thetadot,
+    (I1 + m*h**2)*thetadot.diff(t) +
+    (I3 - I2 - m*h**2)*(v*sm.tan(delta)/b)**2*sm.sin(theta)*sm.cos(theta) -
+    m*g*h*sm.sin(theta) +
+    m*h*sm.cos(theta)*(a*v/b/sm.cos(delta)**2*deltadot +
+                       v**2/v*sm.tan(delta)),
+    x.diff(t) - v*sm.cos(psi),
+    y.diff(t) - v*sm.sin(psi),
+    psi.diff(t) - v/b*sm.tan(delta),
+    delta.diff(t) - deltadot,
```

Comment:
this will display as $\dot{\delta} - \dot{\delta}$, not clear mathematically, should probably introduce other variables.